### PR TITLE
Mylin/#187 improve spectral line query

### LINKED ISSUE #187: Improve ICD test reliability: spectral line query
in `src/test/SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts` a few tests sometimes fail on MacOS and Ubuntu

It looks like it may be something related to this:

```
    Expected: "CH3CHNH2COOH-I"
    Received: "gGG'g-CH2OHCH2CH2OH"

    Expected: undefined
    Received: "CH3CHNH2COOH-I"
```

### PR DESCRIPTION
**Pull request by acdo2002**
Fixed #187 
Each spectral line query has its own session, then it could prevent the sequence disorder. 
Improve the OpenFile to fit the upgrade CLIENT.ts

### CODE REVIEW

File: src/test/CLIENT.ts [hunk 1]
```diff
@@ -515,6 +516,8 @@ export class Client {
                     case CARTA.MomentResponse:
                         ack.MomentResponse.push(data);
                         break;
+                    case CARTA.SpectralLineResponse:
+                        ack.SpectralLineResponse.push(data);
```
Comment (zarda) [inline src/test/CLIENT.ts:520]: Should have a 'break' in the end of case.

Comment (zarda) [reply src/test/CLIENT.ts:520]: The client.stream() might need the similar codes.

File: src/test/SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts [hunk 2]
```diff
@@ -74,52 +74,51 @@ async function delay(milliseconds: number) {
...
-        expect(Connection.connection.readyState).toBe(WebSocket.OPEN);
-    });
+        test(`(Step 0) Connection open? | `, () => {
+            expect(Connection.connection.readyState).toBe(WebSocket.OPEN);
+        });
 
-    assertItem.setSpectralLineReq.map((request, index) => {
-        describe(`(Case ${index}: Line intensity lower limit = ${assertItem.setSpectralLineReq[index].lineIntensityLowerLimit}) between 420GHz and 440GHz |`, () => {
+        describe("Query the spectral line directly with a wide freq range:", () => {
             let response: CARTA.SpectralLineResponse;
```
Comment (zarda) [inline src/test/SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts:92]:
```
let response: AckStream;
```

**Comment by ajm-asiaa** [comment]
Only this part in SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts doesn't seem to consistently pass, even after increasing "spectralLineRequestWide" to 40000.

 ✕ (Step 1) Sent & return SPECTRAL_LINE_RESPONSE within 40000ms (40007ms)
But sometimes it passes quickly:

✓ (Step 1) Sent & return SPECTRAL_LINE_RESPONSE within 40000ms (15661ms)
✓ (Step 1) Sent & return SPECTRAL_LINE_RESPONSE within 30000ms (29570ms)
Could it be a slow network sometimes? I'm not sure how it could be fixed.

**Review by zarda — CHANGES_REQUESTED** [review]
SPECTRAL_LINE_QUERY_WIDE_FREQ sometimes fails, which might relate to internet bandwidth.

**Comment by acdo2002** [comment]
@zarda I have modify these two files based on your suggestions!

**Review by ajm-asiaa — APPROVED** [review]
`SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts` seems to run more reliably now. It passed in 5 out of 5 consecutive tests.

**Review by zarda — APPROVED** [review]

**Pull request merged**